apps/hash-agents/docker/Dockerfile

```diff
@@ -22,28 +22,42 @@ RUN yarn install --frozen-lockfile --prefer-offline && \
 COPY --from=base /app/out/full/ .
 
 
-FROM installer AS builder
+FROM python:3.11-alpine as base-env
 
-WORKDIR /usr/local/src/apps/hash-agents
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 \
+    WORKDIR="/usr/local/src/hash/apps/hash-agents"
+ENV VENV_DIR="$WORKDIR/.venv"
+ENV PATH="$POETRY_HOME/bin:$VENV_DIR/bin:$PATH"
 
-RUN yarn build
+WORKDIR $WORKDIR
 
 
-FROM python:3.11-alpine as runner
+FROM base-env AS builder
 
-RUN apk add --no-cache g++ musl-dev python3-dev libffi-dev make curl
+COPY --from=installer $WORKDIR/poetry.lock $WORKDIR/pyproject.toml ./
 
-WORKDIR /usr/local/src/apps/hash-agents
+RUN apk add --no-cache --virtual .build-deps g++ musl-dev libffi-dev make curl && \
+    curl -sSL https://install.python-poetry.org | python && \
+    poetry install --no-root --only main && \
+    poetry add 'gunicorn[gevent]' && \
+    apk del .build-deps
 
-COPY --from=builder /usr/local/src/apps/hash-agents/requirements.txt ./requirements.txt
-RUN --mount=type=cache,target=/root/.cache/pip \
-    pip3 install -r requirements.txt
 
-COPY --from=builder /usr/local/src/apps/hash-agents .
+FROM base-env AS runner
+
+WORKDIR $WORKDIR
+
+COPY --from=builder $VENV_DIR $VENV_DIR
+COPY --from=installer $WORKDIR/app app
```

Comment:
The gunicorn command won't work if I see this right because the venv is not activated. I might be wrong tho.

Reply:
It is working as `VENV_DIR/bin` is in the path.

Reply:
I see, that makes sense. Thanks for the clarification! It might make sense tho also to set the following env variables:

```
VIRTUAL_ENV=$(poetry env info --path)
POETRY_ACTIVE=1
PATH="$PATH:$VIRTUAL_ENV/bin"
```

(while not required, `VIRTUAL_ENV` is often set for compatibility)

Reply:
Setting the `VIRTUAL_ENV` does not work in this situation where we have a multi-stage build. I renamed `VENV_DIR` to `VIRTUAL_ENV` and added `POETRY_ACTIVE`.